Content.Server/Power/EntitySystems/ApcSystem.cs

```diff
@@ -158,10 +139,15 @@ private void OnEmagged(EntityUid uid, ApcComponent comp, ref GotEmaggedEvent arg
 
             var netBattery = Comp<PowerNetworkBatteryComponent>(uid);
             float power = netBattery is not null ? netBattery.CurrentSupply : 0f;
-
-            if (_userInterfaceSystem.GetUiOrNull(uid, ApcUiKey.Key, ui) is { } bui)
+            if (_ui.TryGetUi(uid, ApcUiKey.Key, out var bui, ui))
             {
-                bui.SetState(new ApcBoundInterfaceState(apc.MainBreakerEnabled, apc.HasAccess, (int)MathF.Ceiling(power), apc.LastExternalState, battery.CurrentCharge / battery.MaxCharge));
+                var state = new ApcBoundInterfaceState(
```

Comment:
FYI there is also a TrySetState(), but its not all that important to change because #15337 will do it anyways just cause I also stumbled on this method while working on that PR